tests/describe_task_queue.go

```diff
@@ -122,42 +130,105 @@ func (s *DescribeTaskQueueSuite) publishConsumeWorkflowTasksValidateBacklogInfo(
 
 	var resp *workflowservice.DescribeTaskQueueResponse
 	var err error
 
-	s.Eventually(func() bool {
-		resp, err = s.engine.DescribeTaskQueue(ctx, &workflowservice.DescribeTaskQueueRequest{
-			Namespace:              s.namespace,
-			TaskQueue:              &taskqueuepb.TaskQueue{Name: tl, Kind: enumspb.TASK_QUEUE_KIND_NORMAL},
-			ApiMode:                enumspb.DESCRIBE_TASK_QUEUE_MODE_ENHANCED,
-			Versions:               nil, // default version, in this case unversioned queue
-			TaskQueueTypes:         nil, // both types
-			ReportPollers:          true,
-			ReportTaskReachability: true,
-			ReportStats:            true,
-		})
-		s.NoError(err)
-		s.NotNil(resp)
-		s.Assert().Equal(1, len(resp.GetVersionsInfo()), "should be 1 because only default/unversioned queue")
-		versionInfo := resp.GetVersionsInfo()[""]
-		s.Assert().Equal(enumspb.BUILD_ID_TASK_REACHABILITY_REACHABLE, versionInfo.GetTaskReachability())
-		types := versionInfo.GetTypesInfo()
-		s.Assert().Equal(len(types), len(expectedBacklogCount))
-
-		validator := true
-		for qT, t := range types {
-			queueType := enumspb.TaskQueueType(qT)
-			if t.Stats.ApproximateBacklogCount != expectedBacklogCount[queueType] {
-				validator = false
+	if isEnhancedMode {
+		s.Eventually(func() bool {
+			resp, err = s.engine.DescribeTaskQueue(ctx, &workflowservice.DescribeTaskQueueRequest{
+				Namespace:              s.namespace,
+				TaskQueue:              &taskqueuepb.TaskQueue{Name: tl, Kind: enumspb.TASK_QUEUE_KIND_NORMAL},
+				ApiMode:                enumspb.DESCRIBE_TASK_QUEUE_MODE_ENHANCED,
+				Versions:               nil, // default version, in this case unversioned queue
+				TaskQueueTypes:         nil, // both types
+				ReportPollers:          true,
+				ReportTaskReachability: true,
+				ReportStats:            true,
+			})
+			s.NoError(err)
+			s.NotNil(resp)
+			s.Assert().Equal(1, len(resp.GetVersionsInfo()), "should be 1 because only default/unversioned queue")
+			versionInfo := resp.GetVersionsInfo()[""]
+			s.Assert().Equal(enumspb.BUILD_ID_TASK_REACHABILITY_REACHABLE, versionInfo.GetTaskReachability())
+			types := versionInfo.GetTypesInfo()
+			s.Assert().Equal(len(types), len(expectedBacklogCount))
+
+			validator := true
+			for qT, t := range types {
+				queueType := enumspb.TaskQueueType(qT)
+				backlogCounter := t.Stats.ApproximateBacklogCount
+				backlogHeadCreateTime := t.Stats.ApproximateBacklogAge.AsDuration()
+				backlogAddTasksRate := t.Stats.TasksAddRate
+				backlogDispatchTasksRate := t.Stats.TasksDispatchRate
+
+				validator = validator &&
+					s.validateBacklogCount(backlogCounter, expectedBacklogCount, queueType) &&
+					s.validateBacklogHeadCreateTime(queueType, backlogHeadCreateTime, nullBacklogHeadCreateTime) &&
+					s.validateAddDispatchTasksRate(queueType, workflows, backlogAddTasksRate, backlogDispatchTasksRate, polled)
 			}
-		}
-		return validator == true
-	}, 3*time.Second, 50*time.Millisecond)
+			return validator == true
```

Comment:
now that I see this whole method, I think it would be a lot simpler if you used EventuallyWithT and then had the validate* functions actually call s.Fail in the false cases

Reply:
I think for now, I want to keep this as is since both the methods perform the same functionality. Can alter this if time permits as I would like to get this in by today and keep an eye out for any potential breaks on main.